test/e2e/user_workload_monitoring_test.go

```diff
@@ -1181,6 +1185,78 @@ func assertEnforcedTargetLimit(limit uint64) func(*testing.T) {
 	}
 }
 
+func assertEnforcedLabelLimit(limit uint64) func(*testing.T) {
```

Comment:
These new assertions aren't called in code, we might want to add them to the existing configuration assertions in the tests.